src/AutoRest.CSharp/Common/Input/InputTypes/InputModelType.cs

```diff
@@ -42,6 +43,8 @@ internal void SetBaseModel(InputModelType? baseModel, [CallerFilePath] string fi
 
         public IEnumerable<InputModelType> GetAllBaseModels() => EnumerateBase(BaseModel);
 
+        public IReadOnlyList<InputModelType> GetImmediateBaseModels() => EnumerateBase(BaseModel, "AzureResourceBase").ToArray();
```

Comment:
this looks very like a workaround, what scenario does this method serve?

Reply:
oh, this is for typespec input, we should not have them in this PR